src/main/java/swbot/InputHandler.java

```diff
@@ -0,0 +1,206 @@
+package swbot;
+import java.util.ArrayList;
+
+/**
+ * A class that handles all the inputs given by the user and parses through them switching between
+ * different commands from the user.
+ */
+public class InputHandler {
+    private ArrayList<Task> data;
+    private Storage storage;
+    private String hline = "____________________________________________________________";
+
+    /**
+     * Creates an inputHandler object that will take care of all the user inputs
+     *
+     * @param data data to be sent to the output and where the chatbot stores the tasks
+     * @param storage storage object which takes care of loading and saving tasks to the database
+     */
+    public InputHandler(ArrayList<Task> data, Storage storage) {
+        this.data = data;
+        this.storage = storage;
+    }
+
+    /**
+     * Handles the majority of the user inputs through other functions in the InputHandler class
+     *
+     * @param input user input that is being passed to the chatbot
+     * @throws BuzzException if any of the commands given by the user is not a valid one
+     */
+    public void overallHandler(String input) throws BuzzException {
+        if (input.startsWith("mark")) {
+            this.markHandle(input);
+            this.storage.saveTasks(data);
+        } else if (input.startsWith("unmark")) {
+            this.unmarkHandle(input);
+            this.storage.saveTasks(data);
+        } else if (input.startsWith("delete")) {
+            this.deleteHandle(input);
+            this.storage.saveTasks(data);
+        } else if (input.equals("list")) {
+            System.out.println(hline);
+
+            for (int i = 0; i < data.size(); i++) {
+                System.out.println((i + 1) + "." + data.get(i).toString());
+            }
+
+            System.out.println(hline);
+        } else if (input.startsWith("todo")) {
+            this.todoHandle(input);
+            this.storage.saveTasks(data);
+        } else if (input.startsWith("deadline")) {
+            this.deadlineHandle(input);
+            this.storage.saveTasks(data);
+        } else if (input.startsWith("event")) {
+            this.eventHandle(input);
+            this.storage.saveTasks(data);
+        } else if (input.startsWith("find")) {
+            this.findHandle(input);
+            this.storage.saveTasks(data);
+        } else {
```

Comment:
Instead of using `if-else` statements, could we consider using a `switch` statement to improve readability?